internal/alloycli/cmd_run.go

```diff
@@ -252,6 +252,8 @@ func (fr *alloyRun) Run(configPath string) error {
 		ClusterName:         fr.clusterName,
 		//TODO(alloy/#1274): graduate to GA once we have more confidence in this feature
 		EnableStateUpdatesLimiter: fr.minStability.Permits(featuregate.StabilityPublicPreview),
+		//TODO(alloy/#1274): graduate to GA once we have more confidence in this feature
+		EnableDiscoveryV2: fr.minStability.Permits(featuregate.StabilityPublicPreview),
```

Comment:
I wonder if we could use something else to control it, like some kind of a temporary feature flag. So that users can still have public preview, but with the old clustering peer discovery if they wanted to.

Reply:
I'm leaning towards changing this to an opt-out flag `--cluster.use-peer-discovery-v1` so that we can disentangle the public preview from clustering peer discovery mechanism.

Reply:
As discussed with the maintainers, I've switched to use a `cluster.use-discovery-v1` flag cc @wildum